### DISCUSSION TITLE
[Feature] Adjust options for slideshow

### DISCUSSION THREAD

**Discussion by tomatonuders** [immich-app/immich#5095] [Feature Request]
### The feature

I was playing around with adding some photos to an album to display photos on an iPad to mimic a digital frame. I would love to be able to adjust the duration of a photo is displayed for. Also, maybe some transition effects like fade in and fade out for photos.  

Another idea is to have a slideshow based on the faces of a person or multiple people. 

The ability to hide the loading bar and the buttons will be nice as well. 

Also noticing that when on the iPad, you cannot select the slideshow option unless you put it on a landscape mode. 

### Platform

- [ ] Server
- [X] Web
- [X] Mobile

**Comment by tomatonuders** [comment]
I'd also like to add that after sharing an album, the slideshow option does not populate.

**Reply by tomatonuders** [reply]
cc @jrasm91 I saw that you were working on the slideshow options. Just wanted your opinion on this.

**Reply by tomatonuders** [reply]
<img width="1498" alt="image" src="https://github.com/immich-app/immich/assets/127623114/5f8220d9-4123-47fc-9041-86a6b4d7a499">

**Reply by jrasm91** [reply]
You mean for a shared link?

**Reply by tomatonuders** [reply]
Is the shared link the same for shared albums? If so they yeah.

**Reply by tomatonuders** [reply]
After generating a Shared link, I also noticed that the options for the slideshow does not populate

**Comment by tomatonuders** [comment]
Shared link also does not have the option for Slideshow

**Comment by mstewart14** [comment]
Options I'd like to see    
1. time a slide is onscreen.  
2. shuffle mode "No Repeats" (no image shows more than once in a complete cycle through the images in the album, this may be dependent on the next option)
3. shuffle mode "Only Once Through" (and then stop, possibly pushing the screen back to the album screen)
4. linear mode "repeat" (cycle through all images linearly indefinitely as shuffle appears to do now)

**Reply by bhugh** [reply]
> linear mode "repeat" (cycle through all images linearly indefinitely as shuffle appears to do now)

+1 to an "infinite loop mode" for forward & backward slideshows.  Since photos come in chronological or reverse chronological order in these slideshows, that is often a feature you do not want to give up by playing them in random shuffle order.  But in many situations you want the slideshow to cycle indefinitely.

**Reply by SimJoSt** [reply]
The shuffle mode should have fewer repeats according to the changelog of the last versions, but there are no different shuffle modes, yet.

For the loop/repeat feature, there is a dedicated feature request for this: https://github.com/immich-app/immich/discussions/18327
It might be better to upvote and comment on that one, for better visibility and more granular consideration and feedback.

**Comment by olibel270** [comment]
Would also love to have an option to hide the blue progress bar on top of the screen.

**Reply by SimJoSt** [reply]
This one has been implemented as an option.

**Reply by olibel270** [reply]
The new options look great! Thank you.

**Comment by Sawtaytoes** [comment]
A lot of these requested features are in. I was wondering if there are more filter settings. Something where I could show pictures with my kids in them or pictures over a certain period of time. Like, I don't care about when I was in college, but I do care to show off pictures of after my kids were born.

Not sure if this is related to something coming in the future:

>  Future updates will aim to understand the temporal and thematic relationships between photos, allowing for even more meaningful slideshows that group similar events or periods together in an organic, surprising way.
>
> Source: https://itsfoss.gitlab.io/post/immich-21-released-with-better-slideshow-shuffle-new-notifications/#introducing-the-revolutionary-better-slideshow-shuffle-in-immich-21

**Reply by Sawtaytoes** [reply]
I think just doing a slideshow of a filtered search page might work:
```search?query=%7B"query"%3A"kids"%2C"takenAfter"%3A"2018-01-01T00%3A00%3A00.000Z"%2C"type"%3A"IMAGE"%7D```

**Comment by Sawtaytoes** [comment]
Because I wanna run my slideshow on an e-ink screen, is there a way to disable videos?

**Reply by Sawtaytoes** [reply]
Yes, the search query does it, and then you can slideshow from there.

**Comment by bo0tzz** [comment]
Most of these options are available today.
